code/modules/vehicles/aircraft.dm

```diff
@@ -0,0 +1,1297 @@
+#define FLYING_CRASH_DAMAGE 20	//Value used for dealing damage when a flying unit bumps/crashes into something
+#define ATTACHED_WEAPON_RELOAD_DELAY 50	//5 seconds	//Time it takes to reload/unload an attached weapon
+#define WEAPON_ATTACH_AND_DETACH_DELAY 50	//5 seconds	//Time it takes to attach/detach a weapon
+#define REFUELING_DELAY 50	//5 seconds	//Time it takes to refuel
+
+/obj/vehicle/sealed/aircraft
+	name = "helicopter"
+	desc = "Fast and nimble with only space for one pilot."
+	icon = 'icons/obj/vehicles/aircraft.dmi'
+	icon_state = ""
+	max_integrity = 50
+	soft_armor = list(MELEE = 0, BULLET = 0, LASER = 0, ENERGY = 0, BOMB = 0, FIRE = 0, ACID = 0)
+	flags_atom = BUMP_ATTACKABLE|PREVENT_CONTENTS_EXPLOSION
+	generic_canpass = FALSE
+	resistance_flags = XENO_DAMAGEABLE|UNACIDABLE|PLASMACUTTER_IMMUNE
+	layer = ABOVE_MOB_LAYER	//So that mobs don't clip above it
+	move_resist = MOVE_FORCE_OVERPOWERING
+	hud_possible = list(MACHINE_HEALTH_HUD, MACHINE_AMMO_HUD)
+	coverage = 100
+	light_system = MOVABLE_LIGHT
+	light_range = 6
+	light_power = 3
+	move_delay = 0.2 SECONDS
+	///How long before the vehicle can change direction
+	var/turn_delay = 0.1 SECONDS
+	///pixel_x and pixel_y offset values; order is clockwise, starting from NORTH
+	var/list/offsets = list()
+	///Total fuel capacity
+	var/max_fuel = 200
+	///How much fuel is in the tank
+	var/current_fuel = 0
+	///How many weapon slots are available
+	var/weapon_slots = 0
+	///List of coordinates to offset gun sprites; every 2 elements are the X and Y, respectively
+	var/list/weapon_slot_positions = list(0, 0)
+	///List of references to displayed gun sprites
+	var/list/weapon_overlays = list()
```

Comment:
doubt you need to be tracking overlays if you use update overlays

Reply:
It's needed to track each specific overlay since you can have multiple weapons attached, else I can't delete X weapon's sprite on the vehicle when it is removed.

Reply:
what? again why isnt this using update overlays